qcfractal/interface/collections/collection.py

```diff
@@ -172,7 +172,11 @@ def from_json(cls, data: Dict[str, Any], client: 'FractalClient' = None) -> 'Col
 
         name = data.pop('name')
         # Allow PyDantic to handle type validation
-        return cls(name, client=client, **data)
+        ret = cls(name, client=client, **data)
+        for optional_field in ["records", "contributed_values"]:
+            if optional_field in ret.data.fields and optional_field not in data:
+                ret.data.__dict__[optional_field] = None
```

Comment:
Somewhat surprised this isn't handled by Pydantic initialization. A bit dangerous to have this so low as `["records", "contributed_values"]` are specific to the Dataset subclasses and not generic Collections.

Reply:
On second thought can we move this to the `Dataset` or `ReactionDataset` class?